### DISCUSSION TITLE
Feed additional info to getStaticProps via getStaticPaths

### DISCUSSION THREAD

**Discussion by simonschwitzky** [vercel/next.js#13489] [Help]
Suppose I have a slug for my website `/en/blog/post-1` matching the dynamic route `/[lang]/blog/[post]`, and I'm using Next.js to export the blog post page at build time using `getStaticProps` and `getStaticPaths`.
The dilemma I am facing is as follows: Inside `getStaticPaths` I query a CMS for all posts, their language and SEO-slug (`[lang]`, `[post]`), to generate an index for the pages and pass these as `params` to `getStaticProps`. Inside, I not want to query only the single post that is being created right now but to do that I need to know the `uid` of the post. I can query that in `getStaticPaths` alongside with the other data, but I haven't yet found a way to pass it into `getStaticProps` since it's not part of the route (and it shouln't be for SEO purposes).
Has anyone experienced something similar or knows of a workaround? Essentially, I am looking for a way to inject data into `context` from within `getStaticPaths`.
Cheers, Simon

**Comment by balazsorban44** [comment]
Might be a duplicate of https://github.com/vercel/next.js/discussions/11272?
